### PR TITLE
HTCONDOR-2081-cgroup-v2-delegation-cleanup

### PR DESCRIPTION
**Pull request by GregThain**
<Insert PR description here, and leave checklist below for code review.>

# HTCondor Pull Request Checklist for internal reviewers

- [ ] Verify that (GitHub thinks) the merge is clean. If it isn't, and you're confident you can resolve the conflicts, do so. Otherwise, send it back to the original developer.
- [ ] Verify that the related Jira ticket exists and has a target version number and that it is correct.
- [ ] Verify that the Jira ticket is in review status and is assigned to the reviewer.
- [ ] Verify that the Jira ticket (HTCONDOR-xxx) is mentioned at the beginning of the title. Edit it, if not
- [ ] Verify that the branch destination of the PR matches the target version of the ticket
- [ ] Check for correctness of change
- [ ] Check for regression test(s) of new features and bugfixes (if the feature doesn't require root)
- [ ] Check for documentation, if needed  (documentation [build logs](https://readthedocs.org/projects/htcondor/builds/))
- [ ] Check for version history, if needed
- [ ] Check BaTLab dashboard for successful build (https://batlab.chtc.wisc.edu/results/workspace.php) and test for either the PR or a workspace build by the developer that has the Jira ticket as a comment.
- [ ] Check that each commit message references the Jira ticket (HTCONDOR-xxx)

## After the above
- Hit the merge button if the pull request is approved and it is not a security patch (security changes require 2 additional reviews)
- If the pull request is approved, take the ticket out of review state
- Assign JIRA Ticket back to the developer

### CODE REVIEW

**Comment by GregThain** [comment]
FWIW, ALICE has been running this branch, and reports that it does what they need.  Their pilots running under a rootly condor can create and manage sub-cgroups to manage the resource that condor give them, and when the condor job is removed, it cleans up all the sub-cgroups and processes therein.

File: src/condor_utils/proc_family_direct_cgroup_v2.cpp [hunk 1]
```diff
@@ -426,61 +513,38 @@ ProcFamilyDirectCgroupV2::kill_family(pid_t pid)
...
-	for (auto entry: stdfs::recursive_directory_iterator{cgroup_mount_point() / cgroup_name}) {
-		if (stdfs::is_directory(entry)) {
-			dirs.emplace_back(entry);
-		}	
+	// This might not exist in your implementation of cgroupv2
+	stdfs::path kill_path = cgroup_mount_point() / stdfs::path(cgroup_name) / stdfs::path("cgroup.kill");
+	FILE *f = fopen(kill_path.c_str(), "r");
+	if (f) {
+		fprintf(f, "%c", '1');
+		fclose(f);
```
Comment (JaimeFrey) [inline src/condor_utils/proc_family_direct_cgroup_v2.cpp:522]: There's a big code clone here with trimCgroupTree(). We should move the "kill everything" bits into their own function.

**Review by JaimeFrey — CHANGES_REQUESTED** [review]
I just noticed that none of this code uses the safe_open() functions. Since this is a virtual FS from the kernel, I think that's ok. Any thoughts on that?

**Review by JaimeFrey — APPROVED** [review]
LGTM

**Pull request merged**